components/admin_console/filter/team_filter_dropdown/team_filter_dropdown.tsx

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {createSelector} from 'reselect';
+
+import {Team} from 'mattermost-redux/types/teams';
+import {debounce} from 'mattermost-redux/actions/helpers';
+
+import LoadingSpinner from 'components/widgets/loading/loading_spinner';
+import InfiniteScroll from 'components/gif_picker/components/InfiniteScroll';
+import * as Utils from 'utils/utils.jsx';
+
+import {FilterOption, FilterValues} from '../filter';
+
+import TeamFilterCheckbox from './team_filter_checkbox';
+
+import './team_filter_dropdown.scss';
+import '../filter.scss';
+
+type Props = {
+    option: FilterOption;
+    optionKey: string;
+    updateValues: (values: FilterValues, optionKey: string) => void;
+
+    teams: Team[];
+    total: number;
+    actions: {
+        getData: (page: number, perPage: number) => Promise<{ data: any }>;
+        searchTeams: (term: string, page?: number, perPage?: number) => Promise<{ data: any }>;
+    };
+};
+
+type State = {
+    page: number;
+    loading: boolean;
+    show: boolean;
+    savedSelectedTeams: Team[];
+    searchResults: Team[];
+    searchTerm: string;
+    searchTotal: number;
+}
+
+const getSelectedTeams = createSelector(
```

Comment:
Do we want to define selectors within the component level? I feel like having these elsewhere may be a better option? I'm not sure if this is something we are already doing or not.

Reply:
This selector is specifically for this component it makes most sense for it to stay here, we already use selectors in a couple of components to memoize results from computed values